internal/engine/netx/errorx/errorx.go

```diff
@@ -244,50 +280,83 @@ func toFailureString(err error) string {
 	if strings.HasSuffix(s, "TLS handshake timeout") {
 		return FailureGenericTimeoutError
 	}
```

Comment:
@kelmenhorst I suppose the above three lines are TLS specific, right? Maybe add a comment mentioning that we should move them or, alternatively, just move them right away?